### PR TITLE
Adds a `gr.DownloadButton` component

### PR DESCRIPTION
**Pull request by abidlabs**
After https://github.com/gradio-app/gradio/pull/7528, I wanted to go through the process of creating a new component with the new approach of working with the files, i.e. setting the URLs in the backend using `serve_static_file()` to mimic the work that a custom component developer would have to do.

I decided to add the long-requested `gr.DownloadButton` component (fixes: https://github.com/gradio-app/gradio/issues/5426)

### CODE REVIEW

**Comment by abidlabs** [comment]
Will probably need to be rebased once https://github.com/gradio-app/gradio/pull/7528 is in

File: demo/fake_gan/files/cheetah1.jpg
Comment (abidlabs) [inline demo/fake_gan/files/cheetah1.jpg:1]: Unused file

File: js/uploadbutton/shared/UploadButton.svelte
Comment (abidlabs) [inline js/uploadbutton/shared/UploadButton.svelte:1]: Fixing the casing in this file

File: gradio/components/download_button.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        icon: str | None = None,
+        scale: int | None = None,
+        min_width: int | None = None,
+        interactive: bool = True,
+        elem_id: str | None = None,
+        elem_classes: list[str] | str | None = None,
+        render: bool = True,
+    ):
+        """
+        Parameters:
```
Comment (dawoodkhan82) [inline gradio/components/download_button.py:43]: Missing param for `icon`

**Comment by dawoodkhan82** [comment]
```
from pathlib import Path
import gradio as gr

def upload_file(filepath):
    name = Path(filepath).name
    return [gr.UploadButton(visible=False), gr.DownloadButton(label=f"Download {name}", value=filepath, visible=True, icon="gradio-logo.png")]

def download_file():
    return [gr.UploadButton(visible=True), gr.DownloadButton(visible=False)]

with gr.Blocks() as demo:
    gr.Markdown("First upload a file and and then you'll be able download it (but only once!)")
    with gr.Row():
        u = gr.UploadButton("Upload a file", file_count="single")
        d = gr.DownloadButton("Download a file", visible=False, icon="gradio-logo.png")

    u.upload(upload_file, u, [u, d])
    d.click(download_file, None, [u, d])


if __name__ == "__main__":
    demo.launch()
```

Tested with this code. And seems like updating the `icon` for the `DownloadButton` doesn't work. Gives error: `AttributeError: 'DownloadButton' object has no attribute 'proxy_url'`.

File: gradio/components/download_button.py (hunk 1 above)
Comment (abidlabs) [reply gradio/components/download_button.py:44]: Added

**Comment by abidlabs** [comment]
Nvm the fix was easier than I thought -- I've merged it into this PR. Your example works for me now @dawoodkhan82

**Review by dawoodkhan82 — APPROVED** [review]
LGTM, tested updating the icon as well. Thanks for including that fix @abidlabs

**Comment by abidlabs** [comment]
Thanks @dawoodkhan82 for testing thoroughly!

**Pull request merged**